### PR TITLE
feat: Adds filter-by-email on /users/_get endpoint for admin calls

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Adds the ability to filter by user emails in the `/users/_get` endpoint for Admin Calls.

From now on, an Admin can query single or multiple users by email when making a `GET` request for `/users` on the instance:

1. For a single user: Simply attach &email=<user1@email.com>
    Eg. `{URL}/users?apiKey=<api_key>&email=teampro@example.com`
2. For multiple users: Simply keep appending &email=<user-n@email.com>
    Eg. `{URL}/users?apiKey=<api_key>&email=teampro@example.com&email=pro@example.com`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: apps/api/pages/api/users/_get.ts [hunk 1]
```diff
@@ -39,6 +51,30 @@ export async function getHandler(req: NextApiRequest) {
   const where: Prisma.UserWhereInput = {};
   // If user is not ADMIN, return only his data.
   if (!isAdmin) where.id = userId;
+
+  if (req.query.email) {
+    const validationResult = schemaQuerySingleOrMultipleUserEmails.safeParse(req.query);
```
Comment (zomars) [inline apps/api/pages/api/users/_get.ts:56]: We can simplify by using parse. `defaultResponder` already handles validations errors.
```suggestion
    const validationResult = schemaQuerySingleOrMultipleUserEmails.parse(req.query);
```

Comment (alishaz-polymath) [reply apps/api/pages/api/users/_get.ts:56]: Oh Cool! 👍

File: apps/api/pages/api/users/_get.ts [hunk 2]
```diff
@@ -39,6 +51,30 @@ export async function getHandler(req: NextApiRequest) {
   const where: Prisma.UserWhereInput = {};
   // If user is not ADMIN, return only his data.
   if (!isAdmin) where.id = userId;
+
+  if (req.query.email) {
+    const validationResult = schemaQuerySingleOrMultipleUserEmails.parse(req.query);
```
Comment (emrysal) [inline apps/api/pages/api/users/_get.ts:56]: This probably should be `safeParse`

**Review by emrysal — CHANGES_REQUESTED** [review]

**Review by emrysal — APPROVED** [review]

File: apps/api/lib/validations/shared/queryUserEmail.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ export const schemaQueryUserEmail = baseApiParams.extend({
 });
 
 export const schemaQuerySingleOrMultipleUserEmails = z.object({
-  email: z.union([z.string().email(), z.array(z.string().email())]).optional(),
+  email: z.union([z.string().email(), z.array(z.string().email())]),
```
Comment (alishaz-polymath) [inline apps/api/lib/validations/shared/queryUserEmail.ts:13]: If we remove optional here, would the endpoint still work if no email is provided?  🤔

Comment (emrysal) [reply apps/api/lib/validations/shared/queryUserEmail.ts:13]: Then req.email is falsy:
image

Comment (alishaz-polymath) [reply apps/api/lib/validations/shared/queryUserEmail.ts:13]: Oh yes, good call!

**Pull request merged**